src/net/KNet/ClientSide/BridgedClasses/Connect/KNetConnectProxy.cs

```diff
@@ -0,0 +1,117 @@
+*
+*  Refer to LICENSE for more information.
+*/
+
+using System;
+
+namespace MASES.KNet.Connect
+{
+    public class KNetConnectProxy
+    {
+        static readonly object globalInstanceLock = new();
+        static KNetConnectProxy globalInstance = null;
+
+        static KNetConnector SinkConnector = null;
+        static KNetConnector SourceConnector = null;
+
+        public static void Register()
+        {
+            lock (globalInstanceLock)
+            {
+                if (globalInstance == null)
+                {
+                    globalInstance = new KNetConnectProxy();
+                    KNetCore.GlobalInstance.RegisterCLRGlobal("KNetConnectProxy", globalInstance);
+                }
+            }
+        }
+
+        public bool AllocateSinkConnector(string connectorClassName)
+        {
+            lock (globalInstanceLock)
+            {
+                try
+                {
+                    var type = Type.GetType(connectorClassName, true);
+                    SinkConnector = Activator.CreateInstance(type) as KNetConnector;
+                    KNetCore.GlobalInstance.RegisterCLRGlobal(SinkConnector.ConnectorName, SinkConnector);
+                    return true;
+                }
+                catch (Exception e)
+                {
+                    return false;
+                }
+            }
+        }
+
+        public bool AllocateSourceConnector(string connectorClassName)
+        {
+            lock (globalInstanceLock)
+            {
+                try
+                {
+                    var type = Type.GetType(connectorClassName, true);
+                    SourceConnector = Activator.CreateInstance(type) as KNetConnector;
+                    KNetCore.GlobalInstance.RegisterCLRGlobal(SourceConnector.ConnectorName, SourceConnector);
+                    return true;
+                }
+                catch (Exception e)
+                {
+                    return false;
+                }
+            }
+        }
+
+        public static void Register<TTask>(KNetSinkConnector<TTask> instance) where TTask : KNetSinkTask
+        {
+            lock (globalInstanceLock)
+            {
+                if (SinkConnector != null) throw new InvalidOperationException("Cannot register multiple KNetSinkConnector");
+                SinkConnector = instance;
+                KNetCore.GlobalInstance.RegisterCLRGlobal(SinkConnector.ConnectorName, SinkConnector);
+            }
+        }
+
+        public static void Register<TTask>(KNetSourceConnector<TTask> instance) where TTask : KNetSourceTask
+        {
+            lock (globalInstanceLock)
+            {
+                if (SourceConnector != null) throw new InvalidOperationException("Cannot register multiple KNetSourceConnector");
+                SourceConnector = instance;
+                KNetCore.GlobalInstance.RegisterCLRGlobal(SourceConnector.ConnectorName, SourceConnector);
+            }
+        }
```

Comment:
Are the functions used?

Reply:
Seems they can be removed. The registration is done automatically when Java requests to allocate the connectors: this is the best way we found to accomplish a start-up in distributed mode.